File: sdks/php/src/Types/ScenarioDownContext.php
<?php

namespace Autonoma\Sdk\Types;

/**
 * Context passed to a scenario's down. Every field is recovered from the
 * verified teardown token, so a down never trusts anything on the request body.
 */
final class ScenarioDownContext
{
    public function __construct(
        /** The scenario name, recovered from the verified teardown token. */
        public readonly string $name,
        /** @var array<string, mixed> The handles this scenario returned from up. */
        public readonly array $teardown,
        /** The testRunId captured at up time. */
        public readonly string $testRunId,
    ) {}
}

File: sdks/java/autonoma-sdk/src/main/java/ai/autonoma/sdk/types/ScenarioDownContext.java
package ai.autonoma.sdk.types;

import java.util.Map;

/**
 * Context passed to a scenario's {@code down}.
 *
 * @param name      the scenario name, recovered from the verified teardown token
 * @param teardown  the {@code teardown} handle this scenario returned from {@code up}
 * @param testRunId the {@code testRunId} captured at {@code up} time
 */
public record ScenarioDownContext(
    String name,
    Map<String, Object> teardown,
    String testRunId
) {}
